ga4gh/backend.py

```diff
@@ -446,137 +455,148 @@ def _noObjectGenerator(self):
         """
-        results = []
-        for obj in self.getDataRepository().getReferenceSets():
-            include = True
-            if request.md5checksum:
-                if request.md5checksum != obj.getMd5Checksum():
-                    include = False
-            if request.accession:
-                if request.accession not in obj.getSourceAccessions():
-                    include = False
-            if request.assembly_id:
-                if request.assembly_id != obj.getAssemblyId():
-                    include = False
-            if include:
-                results.append(obj)
-        return self._objectListGenerator(request, results)
-
-    def referencesGenerator(self, request):
+        query = self._registry_db.get_reference_sets_search_query(request)
+        self._run_sql_query(request, query, responseBuilder)
+
+    def referencesGenerator(self, request, responseBuilder):
         """
         Returns a generator over the (reference, nextPageToken) pairs
         defined by the specified request.
         """
-        referenceSet = self.getDataRepository().getReferenceSet(
-            request.reference_set_id)
-        results = []
-        for obj in referenceSet.getReferences():
-            include = True
-            if request.md5checksum:
-                if request.md5checksum != obj.getMd5Checksum():
-                    include = False
-            if request.accession:
-                if request.accession not in obj.getSourceAccessions():
-                    include = False
-            if include:
-                results.append(obj)
-        return self._objectListGenerator(request, results)
-
-    def variantSetsGenerator(self, request):
+        # Ensure that the reference set exists.
+        self._registry_db.get_reference_set(request.reference_set_id)
+        query = self._registry_db.get_references_search_query(request)
+        self._run_sql_query(request, query, responseBuilder)
+
+    def variantSetsGenerator(self, request, responseBuilder):
         """
         Returns a generator over the (variantSet, nextPageToken) pairs defined
         by the specified request.
         """
-        dataset = self.getDataRepository().getDataset(request.dataset_id)
-        return self._topLevelObjectGenerator(
-            request, dataset.getNumVariantSets(),
-            dataset.getVariantSetByIndex)
+        # Ensure that the dataset exists.
+        self._registry_db.get_dataset(request.dataset_id)
+        query = self._registry_db.get_variant_sets_search_query(request)
+        self._run_sql_query(request, query, responseBuilder)
 
     def variantAnnotationSetsGenerator(self, request):
         """
         Returns a generator over the (variantAnnotationSet, nextPageToken)
         pairs defined by the specified request.
         """
-        compoundId = datamodel.VariantSetCompoundId.parse(
+        compoundId = datamodel.VariantSetCompoundId.parse(  # noqa
```

Comment:
Most of these noqa's are there to keep pep8 happy while the code is incompletely ported. In this example datamodel.VariantSetCompoundId doesn't exist any more, and pep8 freaks out.